src/net/NetworkClient.h

```diff
@@ -5,64 +5,71 @@
 #include <boost/asio/ssl.hpp>
 #include "util/Datagram.h"
 
```

Comment:
Maybe add a comment in this header instructing to subclass NetworkHandler instead of network client?